x-pack/test/functional/services/transform/transform_table.ts

```diff
@@ -214,5 +270,47 @@ export function TransformTableProvider({ getService }: FtrProviderContext) {
       await this.waitForTransformsExpandedRowPreviewTabToLoad();
       await this.assertEuiDataGridColumnValues('transformPivotPreview', column, values);
     }
+
+    public async assertTransformDeleteModalExists() {
+      await testSubjects.existOrFail('transformDeleteModal', { timeout: 60 * 1000 });
+    }
+
+    public async assertTransformDeleteModalNotExists() {
+      await testSubjects.missingOrFail('transformDeleteModal', { timeout: 60 * 1000 });
+    }
+
+    public async assertTransformStartModalExists() {
+      await testSubjects.existOrFail('transformStartModal', { timeout: 60 * 1000 });
+    }
+
+    public async assertTransformStartModalNotExists() {
+      await testSubjects.missingOrFail('transformStartModal', { timeout: 60 * 1000 });
+    }
+
+    public async clickDeleteTransform(transformId: string) {
+      const expectedNumRows = (await this.parseTransformTable()).length - 1;
+
+      await retry.tryForTime(30 * 1000, async () => {
+        await this.assertTransformDeleteModalExists();
+        await testSubjects.click('transformDeleteModal > confirmModalConfirmButton');
+        await this.assertTransformDeleteModalNotExists();
+
+        // If after deletion, and there's no transform left
+        if (expectedNumRows < 1) {
+          await management.assertNoTransformsFoundMessageExists();
+        } else {
+          // Checks that the tranform was deleted
+          await this.filterWithSearchString(transformId, 0);
+        }
+      });
```

Comment:
This retry currently causes test failures because a step in the second part fails and then it retries and expects the model to be open again, which is not the case anymore. I'd suggest to split the functionality:
-  Rename method in `confirmDeleteTransform` and let it stop after `await this.assertTransformDeleteModalNotExists();`
- From the test code wait for the deleted transform is not be displayed anymore (that way we also wouldn't need to pull in the management service here)

Reply:
Updated here https://github.com/elastic/kibana/pull/92314/commits/c6e7f20fa39c45ab29954a1e62c61f88d160d589